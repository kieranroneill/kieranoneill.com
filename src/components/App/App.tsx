import Terminal from '@kieranroneill/terminal-in-react';
import React, { useState } from 'react';
import styled from 'styled-components';

// Commands.
import {
  getAsteroidsCmd,
  getBarrelRollCmd,
  getCVCmd,
  getGithubCmd,
  getLinkedInCmd,
  getTwitterCmd,
  getVersion,
} from '../../commands';

// Components.
import AsteroidsGame from '../AsteroidsGame';
import BarrelRoll from '../BarrelRoll';
import GlobalStyle from '../GlobalStyle';
import Helmet from '../Helmet';

// Descriptions.
import {
  asteroidsDescription,
  barrelrollDescription,
  cvDescription,
  githubDescription,
  linkedinDescription,
  twitterDescription,
  versionDescription,
} from '../../descriptions';

const WrapComponent = styled.div`
  display: flex;
  flex-direction: row;
  height: 100vh;
`;

const App: React.FC = () => {
  const [asteroidsOpen, setAsteroidsOpen] = useState<boolean>(false);
  const [barrelRolling, setBarrelRolling] = useState<boolean>(false);
  const msg: string = `
Welcome to
    __   _                                        _ ____
   / /__(_)__  _________ _____  ____  ____  ___  (_) / /_________  ____ ___
  / // / / _ \\/ ___/ __ \`/ __ \\/ __ \\/ __ \\/ _ \\/ / / // ___/ __ \\/ __ \`__ \\
 / , </ /  __/ /  / /_/ / / / / /_/ / / / /  __/ / / // /__/ /_/ / / / / / /
/_/|_/_/\\___/_/   \\__,_/_/ /_/\\____/_/ /_/\\___/_/_/_(_)___/\\____/_/ /_/ /_/

Type \`help\` to begin'
`;

  return (
    <>
      <GlobalStyle />
      <Helmet />
      {asteroidsOpen && (
        <AsteroidsGame onClose={() => setAsteroidsOpen(false)} />
      )}
      <BarrelRoll
        onComplete={() => setBarrelRolling(false)}
        roll={barrelRolling}
      />
      <WrapComponent>
        <Terminal
          allowTabs={false}
          backgroundColor="black"
          color="green"
          commands={{
            asteroids: getAsteroidsCmd(setAsteroidsOpen),
            barrelroll: getBarrelRollCmd(setBarrelRolling),
            cv: getCVCmd(),
            github: getGithubCmd(),
            linkedin: getLinkedInCmd(),
            twitter: getTwitterCmd(),
            version: getVersion(),
          }}
          descriptions={{
            asteroids: asteroidsDescription,
            barrelroll: barrelrollDescription,
            cv: cvDescription,
            github: githubDescription,
            linkedin: linkedinDescription,
            twitter: twitterDescription,
            version: versionDescription,
          }}
          hideTopBar={true}
          msg={msg}
          promptSymbol="$ "
          showActions={false}
          startState="maximised"
          style={{
            display: asteroidsOpen ? 'none' : 'block',
          }}
        />
      </WrapComponent>
    </>
  );
};

export default App;
