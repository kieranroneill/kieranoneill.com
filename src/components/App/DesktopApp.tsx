import Terminal from '@kieranroneill/terminal-in-react';
import React, { FC, useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

// commands
import getAsteroidsCmd from '@site/commands/asteroids';
import getBarrelRollCmd from '@site/commands/barrelroll';
import getCVCmd from '@site/commands/cv';
import getGithubCmd from '@site/commands/github';
import getLinkedInCmd from '@site/commands/linkedin';
import getVersion from '@site/commands/version';
import getXCmd from '@site/commands/x';

// components
import AsteroidsGame from '@site/components/AsteroidsGame';
import BarrelRoll from '@site/components/BarrelRoll';

// theme
import { palette } from '@site/theme';

// types
import { IBaseCommandOptions, ILogger } from '@site/types';

interface IProps {
  logger: ILogger;
}

const WrapComponent = styled.div`
  display: flex;
  flex-direction: row;
  height: 100vh;
`;

const DesktopApp: FC<IProps> = ({ logger }: IProps) => {
  const { t } = useTranslation();
  // state
  const [asteroidsOpen, setAsteroidsOpen] = useState<boolean>(false);
  const [barrelRolling, setBarrelRolling] = useState<boolean>(false);
  // misc
  const baseCommandOptions: IBaseCommandOptions = {
    logger,
    t,
  };
  const msg: string = `
Welcome to
    __   _                                        _ ____
   / /__(_)__  _________ _____  ____  ____  ___  (_) / /_________  ____ ___
  / // / / _ \\/ ___/ __ \`/ __ \\/ __ \\/ __ \\/ _ \\/ / / // ___/ __ \\/ __ \`__ \\
 / , </ /  __/ /  / /_/ / / / / /_/ / / / /  __/ / / // /__/ /_/ / / / / / /
/_/|_/_/\\___/_/   \\__,_/_/ /_/\\____/_/ /_/\\___/_/_/_(_)___/\\____/_/ /_/ /_/

Type \`help\` to begin'
`;

  return (
    <>
      {asteroidsOpen && (
        <AsteroidsGame onClose={() => setAsteroidsOpen(false)} />
      )}
      <BarrelRoll
        onComplete={() => setBarrelRolling(false)}
        roll={barrelRolling}
      />
      <WrapComponent>
        <Terminal
          allowTabs={false}
          backgroundColor="black"
          color={palette.brand.primary.main}
          commands={{
            asteroids: getAsteroidsCmd({
              ...baseCommandOptions,
              setAsteroidsStateFn: setAsteroidsOpen,
            }),
            barrelroll: getBarrelRollCmd({
              ...baseCommandOptions,
              setBarrelRollStateFn: setBarrelRolling,
            }),
            cv: getCVCmd(baseCommandOptions),
            github: getGithubCmd(baseCommandOptions),
            linkedin: getLinkedInCmd(baseCommandOptions),
            version: getVersion(baseCommandOptions),
            x: getXCmd(baseCommandOptions),
          }}
          descriptions={{
            asteroids: t('commands.asteroids'),
            barrelroll: t('commands.barrelroll'),
            cv: t('commands.cv'),
            github: t('commands.github'),
            linkedin: t('commands.linkedin'),
            version: t('commands.version'),
            x: t('commands.x'),
          }}
          hideTopBar={true}
          msg={msg}
          promptSymbol="$ "
          showActions={false}
          startState="maximised"
          style={{
            display: asteroidsOpen ? 'none' : 'block',
          }}
        />
      </WrapComponent>
    </>
  );
};

export default DesktopApp;
