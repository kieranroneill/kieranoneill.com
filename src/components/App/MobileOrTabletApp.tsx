import { AnimatorGeneralProvider } from '@arwes/animation';
import {
  ArwesThemeProvider,
  Button,
  Card,
  FramePentagon,
  StylesBaseline,
  Text,
} from '@arwes/core';
import React from 'react';
import { useMediaQuery } from 'react-responsive';
import styled from 'styled-components';

// constants
import { Files, Links, Paths, Versions } from '../../constants';

// theme
import { minSizes, palette, typography } from '../../theme';

// utils
import { downloadFile } from '../../utils';

const ButtonContainer = styled.div`
  display: inline-flex;
  flex-direction: column;
  margin: 0 0 1rem;
`;
const Container = styled.div<{ isSmallScreen?: boolean }>`
  ${({ isSmallScreen = false }) => isSmallScreen && 'height: 100%;'}
`;
const Footer = styled.div`
  display: inline-flex;
  flex-direction: column;
`;
const Header = styled(Text)`
  color: ${palette.brand.primary.main};
  text-transform: none;
`;
const StyledButton = styled(Button)`
  margin: 0 0 0.75rem;
  min-width: 12rem;
`;
const Paragraph = styled(Text)`
  margin: 0 0 0.75rem;
`;
const WrapComponent = styled.div`
  align-items: center;
  background-color: ${palette.greyScale.darkerGrey};
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 100vh;
`;

const MobileOrDesktopApp: React.FC = () => {
  const isSmallScreen: boolean = useMediaQuery({ maxWidth: minSizes.tablet });
  const handleDownloadCvClick = () => {
    downloadFile(Paths.DOCS, Files.CV.replace('${version}', Versions.CV));
  };
  const handleLinkClick = (link: string) => () => {
    window.open(link, '_blank');
  };

  return (
    <WrapComponent>
      <ArwesThemeProvider
        themeSettings={{
          palette: {
            primary: palette.brand.primary,
            secondary: palette.brand.secondary,
          },
        }}
      >
        <StylesBaseline
          styles={{
            body: {
              color: palette.brand.primary.main,
              fontFamily: typography.secondaryFontFamily,
            },
          }}
        />
        <Container isSmallScreen={isSmallScreen}>
          <AnimatorGeneralProvider
            animator={{
              duration: {
                enter: 400,
                exit: 400,
                stagger: 80,
              },
            }}
          >
            <Card
              animator={{ activate: true }}
              options={
                <Footer>
                  <ButtonContainer>
                    <StyledButton
                      active={true}
                      FrameComponent={FramePentagon}
                      onClick={handleDownloadCvClick}
                    >
                      <Text>Download CV</Text>
                    </StyledButton>
                    <StyledButton
                      active={true}
                      FrameComponent={FramePentagon}
                      onClick={handleLinkClick(Links.GITHUB)}
                    >
                      <Text>GitHub</Text>
                    </StyledButton>
                    <StyledButton
                      active={true}
                      FrameComponent={FramePentagon}
                      onClick={handleLinkClick(Links.LINKEDIN)}
                    >
                      <Text>LinkedIn</Text>
                    </StyledButton>
                    <StyledButton
                      active={true}
                      FrameComponent={FramePentagon}
                      onClick={handleLinkClick(Links.X)}
                    >
                      <Text>Twitter</Text>
                    </StyledButton>
                  </ButtonContainer>
                  <Text>{`Version ${__VERSION__}`}</Text>
                </Footer>
              }
              style={{
                color: palette.brand.primary.main,
                maxWidth: minSizes.tablet,
                minWidth: minSizes.mobile,
                ...(isSmallScreen && {
                  height: '100%',
                }),
              }}
              title={<Header>Welcome to kieranoneill.com</Header>}
            >
              <Paragraph>
                This is te mobile version of my website where you will find my
                CV and social links.
              </Paragraph>
              <Paragraph>
                Be sure to check out my website on a desktop or laptop, as you
                may find some tasty treats!
              </Paragraph>
            </Card>
          </AnimatorGeneralProvider>
        </Container>
      </ArwesThemeProvider>
    </WrapComponent>
  );
};

export default MobileOrDesktopApp;
