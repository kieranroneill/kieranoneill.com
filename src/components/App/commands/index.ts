export { default as cvCommand } from './cv';
