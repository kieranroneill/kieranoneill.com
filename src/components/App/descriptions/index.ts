export { default as cvDescription } from './cv';
