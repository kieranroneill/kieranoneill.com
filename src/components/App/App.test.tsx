import { shallow } from 'enzyme';
import React from 'react';

// Components.
import App from './App';

describe('components/App', () => {
  it('should mount?', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
