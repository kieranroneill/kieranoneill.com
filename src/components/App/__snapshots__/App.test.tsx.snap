// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App should mount? 1`] = `
<Fragment>
  <Memo(GlobalStyleComponent) />
  <Helmet />
  <styled.div>
    <Terminal
      allowTabs={false}
      backgroundColor="black"
      barColor="black"
      closedMessage="Click on the icon to reopen."
      closedTitle="OOPS! You closed the window."
      color="green"
      commandPassThrough={false}
      commands={Object {}}
      descriptions={Object {}}
      hideTopBar={true}
      msg="Welcome to
    __   _                                        _ ____
   / /__(_)__  _________ _____  ____  ____  ___  (_) / /_________  ____ ___
  / // / / _ \\\\/ ___/ __ \`/ __ \\\\/ __ \\\\/ __ \\\\/ _ \\\\/ / / // ___/ __ \\\\/ __ \`__ \\\\
 / , </ /  __/ /  / /_/ / / / / /_/ / / / /  __/ / / // /__/ /_/ / / / / / /
/_/|_/_/\\\\___/_/   \\\\__,_/_/ /_/\\\\____/_/ /_/\\\\___/_/_/_(_)___/\\\\____/_/ /_/ /_/

Type \`help\` to begin'"
      plugins={Array []}
      prompt="green"
      promptSymbol="$ "
      shortcuts={Object {}}
      showActions={false}
      startState="maximised"
      style={Object {}}
      watchConsoleLogging={false}
    />
  </styled.div>
</Fragment>
`;
